src/bindings/wasm/tvgWgWasmLottieAnimation.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020 - 2024 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <emscripten/bind.h>
+#include <webgpu/webgpu.h>
+#include <thorvg.h>
+#include <iostream>
+
+class TvgWgLottieAnimation
+{
+public:
+    TvgWgLottieAnimation()
+    { 
+        tvg::Initializer::init(0, tvg::CanvasEngine::Wg);
+        canvas = tvg::WgCanvas::gen();
+    }
+
+    ~TvgWgLottieAnimation()
+    {
+        wgpuSurfaceRelease(surface);
+        wgpuInstanceRelease(instance);
+        tvg::Initializer::term();
+     }
+
+    void resize(uint32_t width, uint32_t height)
+    {
+        // instance
+        instance = wgpuCreateInstance(nullptr);
```

Comment:
Do we need new instance creation every resize function called?

Is there another way without recreation? Currently lottie-player calls `resize` every frames.

Reply:
removed as non neccessury